tests/foreman/api/test_contentview.py

```diff
@@ -1167,226 +1239,262 @@ def test_positive_promote_rh_custom_spin(self):
+        role = entities.Role().create()
+        entities.Filter(
+            organization=[self.org],
+            permission=entities.Permission(
+                resource_type='Katello::ContentView').search(
+                filters={'name': 'view_content_views'}),
+            role=role,
+        ).create()
+        # create environment permissions and assign it to our role
+        entities.Filter(
+            organization=[self.org],
+            permission=entities.Permission(
+                resource_type='Katello::KTEnvironment').search(),
+            role=role,
+        ).create()
+        # create a user and assign the above created role
+        entities.User(
+            organization=[self.org],
+            role=[role],
+            login=user_login,
+            password=user_password
+        ).create()
+        # Create new content view entity using admin user
+        content_view = entities.ContentView(organization=self.org).create()
+        cfg = get_nailgun_config()
+        cfg.auth = (user_login, user_password)
+        # Check that we cannot create content view due read-only permission
+        with self.assertRaises(HTTPError):
+            entities.ContentView(cfg, organization=self.org).create()
+        # Check that we can read our content view with custom user
+        content_view = entities.ContentView(cfg, id=content_view.id).read()
+        # Check that we cannot modify content view with custom user
+        with self.assertRaises(HTTPError):
+            entities.ContentView(
+                cfg, id=content_view.id, name=gen_string('alpha')
+            ).update(['name'])
+        # Check that we cannot delete content view due read-only permission
+        with self.assertRaises(HTTPError):
+            content_view.delete()
+        # Check that we cannot publish content view
+        with self.assertRaises(HTTPError):
+            content_view.publish()
+        # Check that we cannot promote content view
+        content_view = entities.ContentView(id=content_view.id).read()
+        content_view.publish()
+        content_view = entities.ContentView(cfg, id=content_view.id).read()
+        self.assertEqual(len(content_view.version), 1)
+        with self.assertRaises(HTTPError):
+            promote(content_view.version[0], lce.id)
 
     @tier2
-    @stubbed()
     def test_negative_non_readonly_user_actions(self):
         """Attempt to view content views
 
-        @id: 85d3f02e-3bd5-4c31-9c64-76a45a3c30a6
+        @id: 9cbc661a-dbe3-4b88-af27-4cf7b9544074
 
-        @setup: create a user withOUT the Content View read-only role
-        @assert: User withOUT read-only role for content view can perform all
-        Variations above
+        @setup: create a user with all Content View permissions except 'view'
+        role
 
-        @caseautomation: notautomated
+        @assert: the user can perform different operations against content
+        view, but not read it
```

Comment:
@oshtaier @abalakh :  I think we can keep this test. Since this is a negative test where user doesn't have read only permission. So user should be able to create but can't read and even if user tries to read, should get an error ( so this is already covered). 

However we should remove other operations like delete etc and keep the test simple as without view permissions, user should get an error on listing.